### PR TITLE
Optimize GitHub actions

### PR DESCRIPTION
**Pull request by SleeplessByte**
Has to be merged after #874.

This PR aims to make it easier to run _much_ more actions on PRs without it slowing down the entire CI.

- Split out configlet to a single action that runs once
- Split out running tests (3 node versions, LTS, previous and next) from linting/stub-checking/etc (LTS only)
- Design new ci scripts `pr` and `pr-check` which _only_ run on exercises that have changed.
- Use the `babel-node` cli, so that the scripts themselves can be written in ES6.

This PR does _not_ sync/update all exercises here in order to keep it a bit more clean. See the next PR.

### CODE REVIEW

File: .github/workflows/ci.precheck.yml
Comment (SaschaMann) [inline .github/workflows/ci.precheck.yml:15]: This workflow doesn't contain a strategy matrix. Is that intended? If so, it's probably better to use a specific version here.

File: .github/workflows/ci.precheck.yml
Comment (SaschaMann) [inline .github/workflows/ci.precheck.yml:18]: Here too

File: .github/workflows/pr.ci.js.yml
Comment (SaschaMann) [inline .github/workflows/pr.ci.js.yml:6]:
```suggestion
on: pull_request
```

I think if you set this to only `pull_request`, you can drop the `if`s below

File: .github/workflows/pr.precheck.yml
Comment (SaschaMann) [inline .github/workflows/pr.precheck.yml:5]:
```suggestion
on: pull_request
```

**Review by SaschaMann — COMMENTED** [review]
I did not check the changes to the scripts, only the workflow files.

File: .github/workflows/ci.precheck.yml
Comment (SleeplessByte) [reply .github/workflows/ci.precheck.yml:15]: No, forgot to put it in after removing them matrix. Thanks!

File: .github/workflows/pr.ci.js.yml
Comment (SleeplessByte) [reply .github/workflows/pr.ci.js.yml:6]: Is there a way where we can still run push, but only in PRs to be merged in?

Push on PRs runs on the post merge code. I think.

Comment (SaschaMann) [reply .github/workflows/pr.ci.js.yml:6]: I'm not sure what you mean. Do you want it to run after the merge button was clicked but before the commit is merged into the main branch?

Comment (SleeplessByte) [reply .github/workflows/pr.ci.js.yml:6]: No, when you do `[push, pull_request]`, you'll normally see your action twice on GitHub.

1. the action will run on the unmerged code (this is what works with the current code)
2. the action will run on a non-existing commit that merges the PR into the target branch.

I'm wondering if we can still achieve number 2, with only the changed files.

Comment (SleeplessByte) [reply .github/workflows/pr.ci.js.yml]: (I've changed it to PR check only for now, since CI check runs after merge, which is fine for now)

File: README.md
Comment (SaschaMann) [inline README.md:2]: This change is a typo, I think?

```suggestion
[![javascript/ci](https://github.com/exercism/javascript/workflows/javascript%20ci/badge.svg)](https://github.com/exercism/javascript/actions?query=workflow%3A%22javascript+ci%22) [![configlet lint](https://github.com/exercism/javascript/workflows/configlet%20lint/badge.svg)](https://github.com/exercism/javascript/actions?query=workflow%3A%22configlet+lint%22) [![CodeQL](https://github.com/exercism/javascript/workflows/CodeQL/badge.svg)](https://github.com/exercism/javascript/actions?query=workflow%3ACodeQL)
```

Comment (SleeplessByte) [reply README.md]: Yes, should have been fixed in the next commit.

I had to push for the workflow to show up before I could change the link 📟

File: README.md
Comment (iHiD) [inline README.md:7]:
```suggestion
This is the JavaScript track, one of the many tracks on [Exercism][web-exercism]. It holds all the _exercises_ that are currently implemented and available for students to complete. The track consists of various **core** exercises - the ones a student _must_ complete - and each **core** exercise may unlock various _side_ exercises. You can find this in the [`config.json`][file-config].
```

File: README.md
Comment (iHiD) [inline README.md:9]:
```suggestion
> In version 3 of Exercism, the above no longer holds. See the [`exercism/v3`][git-v3] repository for more information.
```

**Review by iHiD — APPROVED** [review]

**Pull request merged**
